### PR TITLE
added bookRepo and bookService classes

### PR DESCRIPTION
**Pull request by DmytroHadiuchko**

### CODE REVIEW

File: src/main/java/dmytro/hadiuchko/springboot/entity/Book.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (andrii-hoienko) [inline src/main/java/dmytro/hadiuchko/springboot/entity/Book.java:15]: Please add table annotation

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1,7 @@
 spring.application.name=Spring BOOT
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties]:
```suggestion
```

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,7 @@
 spring.application.name=Spring BOOT
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties]: Don`t push credentials to remote, it is not safe

File: src/main/java/dmytro/hadiuchko/springboot/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (andrii-hoienko) [inline src/main/java/dmytro/hadiuchko/springboot/service/impl/BookServiceImpl.java]:
```suggestion
@Service
@RequiredArgsConstructor
public class BookServiceImpl implements BookService {
    private final BookRepository bookRepository;

```

File: src/main/java/dmytro/hadiuchko/springboot/Application.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package mate.academy.springboot;
+package dmytro.hadiuchko.springboot;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
```
Comment (andrii-hoienko) [inline src/main/java/dmytro/hadiuchko/springboot/Application.java:5]: Requirement: 'Class with @SpringBootApplication annotation that will have a CommandLineRunner bean'. I don`t see these changes

File: pom.xml [hunk 6]
```diff
@@ -33,6 +33,14 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
```
Comment (andrii-hoienko) [inline pom.xml:35]: Did you add checkstyle plugin?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
I don`t see that all requirements are met, please check task description again

File: pom.xml (hunk 6 above)
Comment (DmytroHadiuchko) [reply pom.xml:35]: Yes, it was in first commit, in main branch

https://github.com/DmytroHadiuchko/Spring-BOOT/blob/d760259120ec7fb4f02c096f7da87f49e7d8d470/pom.xml#L42

**Review by karma-o — APPROVED** [review]
Looks okay, but the main problem is that we can not see around half of the changes in that PR, because they are already in the main branch, which is not good. Every change should be merged to main via a PR, especially some business logic like repository/service etc.

**Pull request merged**